### ISSUE TITLE
[Accessibility]: Keyboard navigation issues for notebooks

### ISSUE THREAD

**Issue by novoselrok** [sourcegraph/sourcegraph#35179]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34197

### Problem description

On the notebooks list page:

* Pressing space when a notebooks tab is selected should navigate to that tab

On the notebook edit page:

* Search results in query blocks obstruct the focus and you have to tab through all of them before getting to the menu.
* Open in a new tab menu item is a link, but it looks like a button, so it should react to the space button

### Expected behavior

On the notebooks list page:

* Pressing space when a notebooks tab is selected should navigate to that tab

On the notebook edit page:

* Search results in query blocks obstruct the focus and you have to tab through all of them before getting to the menu.
* Open in a new tab menu item is a link, but it looks like a button, so it should react to the space button

### Additional details

/

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by limitedmage** [comment]
> Search results in query blocks obstruct the focus and you have to tab through all of them before getting to the menu.
I don't think this one is such a big deal as you can use the arrow keys and then shift-tab to go to the previous block's menu. For screen reader users, they can navigate to the menu via the forms list. If you have a better idea on how to do this for keyboard-but-not-screen-reader users let me know.

I'll go ahead and fix the other issues.

**Comment by gitstart-sourcegraph** [comment]
Hi @novoselrok
With regards to second requirement `When focused via tab key, they should become visible`, could you help to validate / invalidate assumption on expected behavior.

### Assumption:
In Notebook view, when menu items which are always visible (`Run search` and `Open in new tab`) are focused, the block should also become focused and as a result show the remaining menu items (`Duplicate`, `Move up`...).

CC: @limitedmage @umpox

**Comment by limitedmage** [comment]
> In Notebook view, when menu items which are always visible (`Run search` and `Open in new tab`) are focused, the block should also become focused and as a result show the remaining menu items (`Duplicate`, `Move up`...).

Yes, I think this is an appropriate assumption. Ideally, focusing on a block and then doing Shift+Tab should also focus on the last menu item of the previous block, even if that menu item is initially hidden. (Hope that makes sense?)
